tests/fields.test.js

```diff
@@ -68,18 +78,41 @@ describe('dataArrayToObject', () => {
     const dataArray = ['5.85', '21.25', '', '', '', '5; asdf; 18'];
     const dataObject = dataArrayToObject(dataArray, fields);
     expect(dataObject).toEqual({
+      a: '5.85',
       b: 21.25,
-      f: [5, 18],
+      f: [5, 'asdf', 18],
     });
   });
 
-  test('includes all values when strict = false', () => {
+  test('includes all values when parseFailureBehavior = REMOVE', () => {
     const dataArray = ['5.85', '21.25', '', '', '', '5; asdf; 18'];
-    const dataObject = dataArrayToObject(dataArray, fields, { strict: false });
+    const dataObject = dataArrayToObject(dataArray, fields, {
+      parseFailureBehavior: REMOVE,
+    });
     expect(dataObject).toEqual({
-      a: '5.85',
-      b: '21.25',
-      f: ['5', 'asdf', '18'],
+      b: 21.25,
+      f: [5, 18],
+    });
+  });
+
+  test('returns correctly formatted string with dateBehavior = JSON_SCHEMA_FORMAT', () => {
```

Comment:
I see now the JSON_SCHEMA_FORMAT has to be defined in order to handle conversion from internal Javascript date object out to JSON (using XML standard date/time types).  And then you allow custom formats too for date, datetime, and time.